design/proposal-updateable-specs.md

```diff
@@ -0,0 +1,90 @@
+
+It is also difficult to justify the tight coupling of Troubleshoot configuration, which is intended to be an entirely client-side concept, and the application manager which is installed to a Kubernetes environment.
+
+## Goals
+
+* Provide the ability to automatically obtain a spec from a source that receives updates, whenever Troubleshoot is run
+* Decouple the provision of default specs from the kots application manager, so that kots upgrades are not required in order to get access to updated specs
+
+## Non Goals
+
+* Maintain the ability for Troubleshoot to run in an airgapped environment
+* maintain compatilbility with existing Troubleshoot specs
+
+## High-Level Design
+
+Add a new field to the Troubleshoot spec definition, which includes a URI used to locate the current spec online.
+
+If the field is populated, Troubleshoot is to attempt to download the additional spec(s) from the source online, and add to the spec already provided.
+
+Add a CLI flag to prevent any attmept to download, for use in airgap environments and/or when we simply do not want to download.
+
+Update the [default spec included with kots](https://github.com/replicatedhq/kots/blob/main/pkg/supportbundle/defaultspec/spec.yaml) to include a URI pointing at [the troubleshoot-specs repo](https://raw.githubusercontent.com/replicatedhq/troubleshoot-specs/main/in-cluster/default.yaml).
+
+URI's provided may be any of the URI types already accepted by Troubleshoot, including web addresses (https://), secrets, or files.
+
+## Detailed Design
+
+Current spec format is like this example:
+
+```yaml
+apiVersion: troubleshoot.sh/v1beta2
+kind: SupportBundle
+metadata:
+  name: default
+spec:
+  collectors:
+    - clusterInfo: {}
+    - clusterResources: {}
+  analyzers:
+    - cephStatus: {}
+    - longhorn: {}
+```
+
+We could add a new type, `specURIs`, which contains a list of URIs from which to obtain specs:
+
+```yaml
+apiVersion: troubleshoot.sh/v1beta2
+kind: SupportBundle
+metadata:
+  name: default
+spec:
+  specURIs:
+    - https://raw.githubusercontent.com/replicatedhq/troubleshoot-specs/main/in-cluster/default.yaml
+  collectors:
+    - clusterInfo: {}
+    - clusterResources: {}
+  analyzers:
+    - cephStatus: {}
+    - longhorn: {}
+```
+
+## Impact on kURL and kots
+
+For kURL, each component/addon installed could include adding a URI pointing at a specific spec that 
```

Comment:
oops.

For kURL, each component/addon installed could include adding a URI pointing at a specific spec for that component, which could be updated independently of other components.